### ISSUE TITLE
understanding errors: is it possible to view the http request and responses on getToken()?

### ISSUE THREAD

**Issue by kenirwin** [lelylan/simple-oauth2#284]
I'm trying to use simple-auth2 to get an access token, where then clientId and clientSecret are hard-coded in my app. I'm getting the "content-type is not JSON compatible" error. I suspect that it's because the API I'm connecting to doesn't like the way the token request is being submitted, and it's sending back an HTML page with the errors. 

So: 
* Is there a way to see what the token-request HTTP call looks like so I can debug my settings? 
* Is there a way to see what the response from the API is so I can read their messages?

I'm using:
```
const oauth2 = require('simple-oauth2').create(credentials); // I've set the credentials object
app.get('/callback', async (req,res) => {
  try {
    const result = await oauth2.authorizationCode.getToken();
    const accessToken = oauth2.accessToken.create(result);
    console.log('Token: ', result);
  } catch (err) {
    console.error('Access Token Error', err.message);
  }
})
```

I know that I can get a token back from the API by making a POST call with
data: `grant_type=client_credentials&client_id=<id>&client_secret=<secret>`

To accomplish this, I'm using 
```
credentials = {
  client: {
    id: '<>',
    secret: '<>',
    secretParamName: 'client_secret',
    idParamName: 'client_id'
  },
  auth: {
// a bunch of auth definitions
  },
  http: {
    headers: {
      grant_type: 'client_credentials',
    }
  },
  options: {
    bodyFormat: 'form'
  }
}
```

Can you point me in the right direction, either to read the debugging data (http post request and the API's response) or in tailoring my credentials object to match the known POST format for a working request?

Thanks!

**Comment by jaggygit** [comment]
The error is addressed #279 and its answered by @korneliuscode. You should add the following to your credentials
`http: {
	json: 'force'
}`

**Comment by kenirwin** [comment]
That worked -- thank you!
